native/cocos/core/assets/SimpleTexture.cpp

```diff
@@ -159,23 +159,24 @@ void SimpleTexture::createTexture(gfx::Device *device) {
     notifyTextureUpdated();
 }
 
-void SimpleTexture::createTextureView(gfx::Device *device) {
+gfx::Texture *SimpleTexture::createTextureView(gfx::Device *device) {
     if (!_gfxTexture) {
-        return;
+        return nullptr;
     }
+    const uint32_t maxLevel = _maxLevel < _mipmapLevel ? _maxLevel : _mipmapLevel - 1;
     auto textureViewCreateInfo = getGfxTextureViewCreateInfo(
     _gfxTexture,
     getGFXFormat(),
     _baseLevel,
-    _maxLevel - _baseLevel + 1
+    maxLevel - _baseLevel + 1
     );
 
     //TODO(minggo)
 //    if (!textureViewCreateInfo) {
 //        return;
 //    }
     
-    _gfxTextureView = device->createTexture(textureViewCreateInfo);
+    return device->createTexture(textureViewCreateInfo);
```

Comment:
We should also free their memory when destroying `_gfxTexture` and `_gfxTextureView`.

Reply:
It is destroyed in `SimpleTexture::tryDestroyTextureView`. And it is the same logic as TS.

Reply:
Yes, exactly, Typescript has a GC system, and the memory is freed automatically. But for C++, things are different, `_gfxTexture->destroy()` will only free the resource on GPU, and we need to call `CC_DELETE(_gfxTexture)` mannually.

Reply:
They are IntrusivePtr, destructor will be invoked automatically.

Reply:
Ok, fine.